qa/acceptance/spec/shared_examples/cli/logstash-plugin/update.rb

```diff
@@ -33,23 +33,39 @@
 
     before do
       logstash.run_command_in_path("bin/logstash-plugin install --no-verify --version #{previous_version} #{plugin_name}")
+      logstash.run_command_in_path("bin/logstash-plugin list")
       # Logstash won't update when we have a pinned version in the gemfile so we remove them
```

Comment:
this comment appears to go with a line that was moved.

It is not clear to me _why_ we need to do a gemfile replacement to un-pin, and whether that is supposed to affect the result of the `have_installed` validation that is now occurring before the un-pinning.

Reply:
I've moved the comment around.

It's a pretty hacky test. We install a named, older version of the test plugin, and do a `have_installed` check  to ensure the pre-condition that the plugin has installs as expected. The `sed` ugliness that is done on the Gemfile is to remove the pin - `bin/logstash-plugin update` will not upgrade pinned plugins (in this case a plugin that has been installed at a specific version). Whether that is desired behaviour, or whether it is something we should allow to be bypassed via a `--force` option is a valid question